Sources/MapboxNavigation/Route.swift

```diff
@@ -42,4 +42,62 @@ extension Route {
             return LineString(trimmedPrecedingCoordinates + followingPolyline.trimmed(from: followingPolyline.coordinates[0], distance: distance)!.coordinates.suffix(from: 1))
         }
     }
+
+    var tollIntersections: [Intersection]? {
+        let allSteps = legs.flatMap { return $0.steps }
+
+        let allIntersections = allSteps.compactMap { return $0.intersections }.reduce([], +)
```

Comment:
```suggestion
        let allIntersections = allSteps.flatMap { $0.intersections ?? [] }
```
this will work in one pass, instead of two, a little bit faster🙂